test/minikube/001-kube.bats

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bats
+#
+# Tests of podman kube commands with minikube
+#
+
+load ../system/helpers.bash
+
+function setup(){
+    # only set up the minikube cluster before the first test
+    if [[ "$BATS_TEST_NUMBER" -eq 1 ]]; then
+        minikube start
+        echo $output >&2
+    fi
+    # wait for a bit so all the services in the cluster come up
+    sleep 10
+}
+
+function wait_for_pods_to_start(){
+    count=0
+    running=false
+    # timeout after 30 seconds
+    # if the pod hasn't started running after 30 seconds, there is something else wrong
+    while [[ $count -lt 30 ]] && [[ $running == false ]]
+    do
+        run minikube kubectl get pods
+        assert "$status" -eq 0
+        if [[ "$output" =~ "Running" ]]; then
+            running=true
+            count=31
+        fi
+        count=$((count + 1))
+        sleep 1
+    done
+    if [[ $running == false ]]; then
+        echo "Timed out waiting for pod to move to 'Running' state"
+        exit 1
+    fi
+}
+
+###############################################################################
+# BEGIN tests
+
+@test "minikube - check cluster is up" {
+    run minikube kubectl get nodes
+    assert "$status" -eq 0
+    assert "$output" =~ "Ready"
+    run minikube kubectl get pods
+    assert "$status" -eq 0
+    assert "$output" == "No resources found in default namespace."
+}
+
+@test "minikube - deploy generated container yaml to minikube" {
+    cname="test-ctr"
+    fname="/tmp/minikube_deploy_$(random_string 6).yaml"
+    run_podman container create --name $cname $IMAGE top
+    run_podman kube generate -f $fname $cname
+
+    # deploy to the minikube cluster
+    project="ctr-ns"
+    run minikube kubectl create namespace $project
+    assert "$status" -eq 0
+    run minikube kubectl -- apply -f $fname
+    assert "$status" -eq 0
+    assert "$output" == "pod/$cname-pod created"
+    wait_for_pods_to_start
+    assert $status -eq 0
```

Comment:
Unnecessary (`wait_for_pods_to_start` dies on error; and it does not set `$status`). Please remove.

Reply:
fixed